src/ray/protobuf/experimental/instance_manager.proto

```diff
@@ -135,6 +135,34 @@ message LaunchRequest {
   int64 request_ts_ms = 4;
 }
 
+// Termination request for an instance.
+message TerminationRequest {
+  enum Cause {
+    UNKNOWN = 0;
+    // Idle termination.
+    IDLE = 1;
+    // Per type max instance limit reached.
+    MAX_NUM_NODE_PER_TYPE = 2;
+    // Global max num nodes reached.
+    MAX_NUM_NODES = 3;
+  }
+  // id of the request.
+  string id = 1;
+  // The ray node id of the instance that's terminated.
+  string ray_node_id = 2;
+  // the instance id of the instance that's terminated.
+  string instance_id = 3;
+  // the reason of the termination.
+  Cause cause = 4;
+
+  // Idle time if the cause is IDLE.
+  optional int64 idle_time_ms = 5;
```

Comment:
1. Does `ms` mean milliseconds? If so, the unit is too granular. We can use the second as the unit.

2. `int64` seems to be an overkill from my perspective. If we still use int32, 

  * (2^31 - 1) ms / 1000 / 60 / 60 = 596.5 hours.

For me, idle time exceeding 72 hours (Friday ~ Monday) is unnecessary.